exec/java-exec/src/main/java/org/apache/drill/exec/record/RawFragmentBatch.java

```diff
@@ -77,4 +83,47 @@ public long getByteCount() {
   public boolean isAckSent() {
     return ackSent.get();
   }
+
+  /**
+   * Transfer ownership of this DrillBuf to the target allocator. This is done for better memory
+   * accounting (that is, the operator should be charged with the body's Drillbuf memory).
+   *
+   * <p><b>NOTES -</b>
+   * <ul>
+   * <li>This operation is a NOOP when a) the current allocator (associated with the DrillBuf) is not the
+   *     owning allocator or b) the target allocator is already the owner
+   * <li>When transfer happens, a new RawFragmentBatch instance is allocated; this is done for proper
+   *     DrillBuf reference count accounting
+   * <li>The RPC handling code caches a reference to this RawFragmentBatch object instance; release()
+   *     calls should be routed to the previous DrillBuf
+   * <li>Caller is responsible for checking an OOM condition
+   * </ul>
+   *
+   * @param targetAllocator target allocator
+   * @return a new {@link RawFragmentBatch} object instance on success (where the buffer ownership has
+   *         been switched to the target allocator); otherwise this operation is a NOOP (current instance
+   *         returned)
+   */
+  public RawFragmentBatch transferBodyOwnership(BufferAllocator targetAllocator) {
```

Comment:
DrillBuf has a transferOwnership method that could be used instead? See the transferTo methods in any of the vector classes -  [FixedValueVectors.transferTo(....)](https://github.com/apache/drill/blob/master/exec/vector/src/main/codegen/templates/FixedValueVectors.java#L282) 
Saves you the trouble of rolling your own and also avoids the need to make changes to any of the existing classes.